scribus/plugins/import/pdf/slaoutput.cpp

```diff
@@ -3896,3 +3926,680 @@ bool SlaOutputDev::checkClip()
+	}
+	bool yInLimits = false;
+	if (closeToY(newPoint.y(), lastXY.y()))
+	{
+		yInLimits = true;
+	}
+	FrameworkLineTests pass = linearTest(newPoint, xInLimits, yInLimits);
+	return pass;
+}
+
+
+TextRegion::FrameworkLineTests TextRegion::moveToPoint(QPointF newPoint)
+{
+	//
+	//qDebug() << "moveToPoint: " << newPoint;
+	// Do some initilization if we are in a new text region
+	// we could also update these if glyphindex = -1;
+	if (glyphs.empty())
+	{
+		lineBaseXY = newPoint;
+		lastXY = newPoint;
+		//qDebug() << "newPoint3: " << newPoint;
+	}
+	
+	//TODO: I need to write down which ones we want so I can work it all out, for now just some basic fuzzy matching support.
+	//TODO: x limiting should be different for moving as opposed to adding a new glyph because moiving is due to a discontinuity in glyphs
+	bool xInLimits = false;
+	if (closeToX(newPoint.x(), lastXY.x()))
+	{
+		//qDebug() << "newPoint6: " << newPoint;
+		xInLimits = true;
+	}
+	bool yInLimits = false;
+	if (closeToY(newPoint.y(), lastXY.y()))
+	{
+		//qDebug() << "newPoint7: " << newPoint;
+		yInLimits = true;
+	}
+	//qDebug() << "newPoint8: " << newPoint;
+	FrameworkLineTests pass = linearTest(newPoint, xInLimits, yInLimits);
+	//TODO: need to check to see if we are creating a new paragraph or not. basically if the cursor is returned to x origin before it reached x width. this could be returned as part of a matrix by linearTest that specifies exactly how the test ws passed. maybew return an enum with either the mode that passed or a failure value
+	if (pass != FrameworkLineTests::FAIL)
+	{
+		//create new lines and line segments depending upon the moide of the movement.
+		TextRegionLine *textRegionLine = nullptr;
+
+		if (pass == FrameworkLineTests::NEWLINE || pass == FrameworkLineTests::FIRSTPOINT)
+		{
+			//qDebug() << "Newline: ";
+			if (pass == FrameworkLineTests::FIRSTPOINT)
+			{
+				if (textRegionLines.empty()) {
+					textRegionLines.push_back(TextRegionLine());
+				}
+			} 
+			else
+			{
+				textRegionLines.push_back(TextRegionLine());
+			}
+			textRegionLine = &textRegionLines.back();
+			textRegionLine->baseOrigin = newPoint;
+			if (pass == FrameworkLineTests::NEWLINE)
+			{
+				textRegionLine->maxHeight = abs(newPoint.y() - lastXY.y());
+				if (textRegionLines.size() == 2)
+				{
+					lineSpacing = abs(newPoint.y() - lastXY.y()) + 1;
+					//qDebug() << "setting lineSpacing to:" << lineSpacing;
+				}
+			}
+		}
+
+		textRegionLine = &textRegionLines.back();
+		if ((pass == FrameworkLineTests::FIRSTPOINT && textRegionLine->segments.empty()) || pass == FrameworkLineTests::NEWLINE ||  pass != FrameworkLineTests::FIRSTPOINT && textRegionLine->segments[0].glyphIndex != textRegionLine->glyphIndex)
+		{
+			TextRegionLine newSegment = TextRegionLine();
+			textRegionLine->segments.push_back(newSegment);
+		}
+		TextRegionLine* segment = &textRegionLine->segments.back();
+		segment->baseOrigin = newPoint;
+		if (pass == FrameworkLineTests::STYLESUPERSCRIPT)

```

Comment:
since you seem to love `? : ` here it might or not help make the code a bit more readable : - )
```
segment->maxHeight = (pass == FrameworkLineTests::STYLESUPERSCRIPT) ?
    abs(lineSpacing - (newPoint.y() - lastXY.y())) :
    textRegionLines.back().maxHeight;
```

i'm not sure that it's really more clear... but at least it's a bit more clear that you're setting `maxHeight`.

in any case, if you stick with the if, remove the curly brackets... this is a case where it even makes sense not to use them...

Reply:
good idea